app/src/main/java/com/example/concordia_campus_guide/Adapters/RoutesAdapter.java

```diff
@@ -0,0 +1,208 @@
+        mainTransportType.setImageResource(R.drawable.ic_directions_car_red);
+
+        TextView summary = new TextView(context);
+        summary.setText(route.getSummary());
+        summary.setGravity(Gravity.CENTER);
+        details.addView(summary);
+
+        duration.setText(route.getDuration());
+    }
+
+    private void setUIBus(Bus bus) {
+        ImageView busIcon = new ImageView(context);
+        busIcon.setImageResource(R.drawable.ic_directions_bus_red);
+
+        TextView busNumber = new TextView(context);
+        busNumber.setText(bus.getBusNumber());
+        busNumber.setGravity(Gravity.CENTER);
+
+        details.addView(busIcon);
+        details.addView(busNumber);
+    }
+
+    private void setUISubway(Subway subway) {
+        ImageView subwayIcon = new ImageView(context);
+        subwayIcon.setImageResource(pickCorrectSubwayColor(subway.getColor()));
+
+        details.addView(subwayIcon);
+    }
+
+    private int pickCorrectSubwayColor(String color) {
+        if(color.equalsIgnoreCase("#0074b9"))
+            return R.drawable.ic_subway_blue;
+        else if(color.equalsIgnoreCase("#f27f2f"))
+            return R.drawable.ic_subway_orange;
+        else if(color.equalsIgnoreCase("#00a556"))
+            return R.drawable.ic_subway_green;
+        else if(color.equalsIgnoreCase("#f2d101"))
+            return R.drawable.ic_subway_yellow;
+
+        return -1;
+    }
+
+    private void setUITrain(Train train) {
+        ImageView trainIcon = new ImageView(context);
+        trainIcon.setImageResource(R.drawable.ic_train_red);
+
+        TextView trainShortName = new TextView(context);
+        trainShortName.setText(train.getTrainShortName());
+        trainShortName.setGravity(Gravity.CENTER);
+
+        details.addView(trainIcon);
+        details.addView(trainShortName);
+    }
+
+    private void setUIArrow() {
+        ImageView arrow = new ImageView(context);
+        arrow.setImageResource(R.drawable.ic_keyboard_arrow_right_grey);
+        details.addView(arrow);
+    }
+
+    private void setUITransit(Route route) {
+        mainTransportType.setImageResource(R.drawable.ic_directions_bus_red);
+        duration.setText(route.getDuration());
+    }
+
+    private void setBottomLayout(Route route) {
+        if(route.getArrivalTime().isEmpty() || route.getDepartureTime().isEmpty())
+            bottom.setVisibility(LinearLayout.GONE);
+        else
+            arrivalAndDepartureTime.setText(route.getDepartureTime() + " - " + route.getArrivalTime());
+    }
+
+    private void displayInfoMsg() {
+        bottom.setVisibility(LinearLayout.GONE);
+        details.setVisibility(LinearLayout.GONE);
+        arrivalAndDepartureTime.setVisibility(LinearLayout.GONE);
+        mainTransportType.setVisibility(LinearLayout.GONE);
+        main.setVisibility(LinearLayout.GONE);
+
+        TextView warningMsg = new TextView(context);
+        warningMsg.setText("No routes available.");
```

Comment:
plz plz put in Strings.xml, it will be easier to move to french by doing that.

Reply:
got it!